server/methods/core/orders.js

```diff
@@ -62,53 +62,115 @@ export function ordersInventoryAdjust(orderId) {
  */
 export const methods = {
   /**
+   * orders/shipmentPicked
+   *
+   * @summary update picking status
+   * @param {Object} order - order object
+   * @param {Object} shipment - shipment object
+   * @return {Object} return workflow result
+   */
+  "orders/shipmentPicked": function (order, shipment) {
```

Comment:
This feels brittle to me as well.

I'm always open to new ways to do things, but it seems like we should have "orders/progressStatus" and "orders/regressStatus" methods instead of one method per status.

We could then have a set of workflows that identifies the next or previous status from the existing status and proceeds. There could be different states for items than for orders, and each should be overridable. 

This seems like it would make it easier to extend or replace functionality because if you replace the workflow set, then you replace the way that orders progress.